lm_eval/tasks/conala.py

```diff
@@ -0,0 +1,103 @@
+split into 2,379 training and 500 test examples.
+
+Homepage: https://conala-corpus.github.io/
+Here we use two-shot evaluation (the original paper evaluates finetuned models)
+"""
+
+import json
+from evaluate import load
+from lm_eval.base import Task
+
+
+_CITATION = """
+@inproceedings{yin2018learning,
+  title={Learning to mine aligned code and natural language pairs from stack overflow},
+  author={Yin, Pengcheng and Deng, Bowen and Chen, Edgar and Vasilescu, Bogdan and Neubig, Graham},
+  booktitle={2018 IEEE/ACM 15th international conference on mining software repositories (MSR)},
+  pages={476--486},
+  year={2018},
+  organization={IEEE}
+}
+"""
+
+
+class Conala(Task):
+    """A task represents an entire benchmark including its dataset, problems,
+    answers, generation settings and evaluation methods.
+    """
+
+    DATASET_PATH = "neulab/conala"
+
+    def __init__(self):
+        super().__init__(
+            stop_words=["\n"],
+            requires_execution=False,
+        )
+
+    def get_dataset(self):
+        """Returns dataset for the task or an iterable of any object, that get_prompt can handle"""
+        return self.dataset["test"]
+
+    def fewshot_examples(self):
+        """Loads and returns the few-shot examples for the task if they exist."""
+        with open(
+            "lm_eval/tasks/few_shot_examples/conala_few_shot_prompts.json", "r"
+        ) as file:
+            examples = json.load(file)
+        return examples
+
+    @staticmethod
+    def two_shot_prompt(entry, text, examples):
+        """Two shot prompt format as instructions & solutions"""
+        prompt = f"\nInstruction:\n{examples['instruction1']}\
+                   \nSolution:\n{examples['solution1']}\
+                   \nInstruction:\n{examples['instruction2']}\
+                   \nSolution:\n{examples['solution2']}\
+                   \nInstruction:\n{text}\
+                   \nSolution:\n"
+        return entry + prompt
+
+    def get_prompt(self, doc):
+        """Builds the prompt for the LM to generate from."""
+        examples = self.fewshot_examples()
+        text_column = "rewritten_intent" if doc["rewritten_intent"] else "intent"
+        text = doc[text_column].strip()
+        entry = "Answer the following instructions in one line of Python code:\n"
+        prompt = self.two_shot_prompt(entry, text, examples)
+        return prompt
+
+    def get_reference(self, doc):
+        """Builds the reference solution for the doc (sample from the test dataset)."""
+        return doc["snippet"]
+
+    def postprocess_generation(self, generation, idx):
+        """Defines the postprocessing for a LM generation.
+        :param generation: str
+            code generation from LM
+        :param idx: int
+            index of doc in the dataset to which the generation belongs
+            (not used for this task)
+        """
+        output = generation.split("Solution:\n", 3)[-1]
+        output = output.split("\n")[0]
```

Comment:
Why do we need `output.split("\n")[0]`?
An example solution from Conala is `mixed.replace('\r\n', '\n').replace('\r', '\n')`, which would get invalidated by this if the model generated it correctly afaict

(same comment for concode)

Reply:
Good catch ,I must've left this from when teh stopping criteria on `\n` wasn't working so I had to take the first generated line. I will replace it with `output.strip()` to remove the stop word  `\n`  if it's included